### DISCUSSION TITLE
How to check if a font has been modified?

### DISCUSSION THREAD

**Discussion by ftCLI** [fonttools/fonttools#2667] [Q&A] [answered]
When performing operations on fonts, is there a simple way to detect if a font has been modified?

```
for f in files:

    font = ttLib.TTFont("font.ttf")

    font["OS/2"].usWeightClass = 300

    if font.modified == True:
        font.save(f)
    else:
        print("No changes made.")
```

I'd like to avoid saving a font if hasn't been modified (in the example, if usWeightClass value was already 300 the font must not be saved)

**Comment by anthrotype** [comment, marked as answer]
No, that's not possible, unless we were to add some magic that intercepts setattr, setitem, etc. throughout the lib to watch for any changes (which I think would be overkill).

**Reply by anthrotype** [reply]
in an ideal world, we would use immutable types everywhere and any change would entail creating a new object (similar to dataclasses.replace or attrs.evolve for frozen classes), but that's also hard to do at this stage. It'd mean a complete rewrite, given how most fonttools stuff relies on object mutability
